salt/modules/win_event.py

```diff
@@ -0,0 +1,754 @@
+"""
+A module for working with the Windows Event log system.
+"""
+# https://docs.microsoft.com/en-us/windows/win32/eventlog/event-logging
+
+import collections
+import logging
+
+import xmltodict
+
+import salt.utils.platform
+import salt.utils.stringutils
+from salt.exceptions import CommandExecutionError
+
+try:
+    import pywintypes
+    import win32evtlog
+    import win32evtlogutil
+    import winerror
+
+    IMPORT_STATUS = True
+except ImportError:
+    IMPORT_STATUS = False
+
+# keys of all the parts of a Event supported by the API
+EVENT_PARTS = (
+    "closingRecordNumber",
+    "computerName",
+    "data",
+    "eventCategory",
+    "eventID",
+    "eventType",
+    "recordNumber",
+    "reserved",
+    "reservedFlags",
+    "sid",
+    "sourceName",
+    "stringInserts",
+    "timeGenerated",
+    "timeWritten",
+)
+
+EVENT_TYPES = {
```

Comment:
We should avoid using mutable objects at the global scope. It looks like this is only being used in one function, can we move this and the other variables here to the functions?